config/paths.js

```diff
@@ -3,12 +3,34 @@
 const path = require('path');
 const fs = require('fs');
 const url = require('url');
+const cosmiconfig = require('cosmiconfig');
 
 // Make sure any symlinks in the project folder are resolved:
 // https://github.com/facebookincubator/create-react-app/issues/637
 const appDirectory = fs.realpathSync(process.cwd());
 const resolveApp = relativePath => path.resolve(appDirectory, relativePath);
 
+// We look for configration in files supported by cosmiconfig by default:
+// https://github.com/davidtheclark/cosmiconfig
+const explorer = cosmiconfig('elmapp');
+const result = explorer.searchSync(appDirectory);
+const config = result ? result.config : loadElmJson();
```

Comment:
If someone introduces the new config file, but forgets to remove the same fields in `elm.json` (and don't run `elm-package` so the fields are actually kept), this might get confusing, as to which file is used (it's obviously the config file, but users won't know that). Maybe print a warning here if there is a config file _and_ `loadElmJson()` returns a non-empty object?